iModelCore/ecobjects/test/NonPublished/ECJsonUtilitiesTests.cpp

```diff
@@ -42,17 +88,17 @@ TEST_F(ECJsonUtilitiesTestFixture, JsonToId)
 
     jsonStr = "-10"; // Not a valid value for an Id, but failing the method is not worth the overhead. So the test documents the behavior of the API
     ASSERT_EQ(SUCCESS, ParseJson(jsonDoc, jsonStr));
-    EXPECT_EQ(BeInt64Id(uint64_t(-10)), ECJsonUtilities::JsonToId<BeInt64Id>(jsonDoc)) << jsonStr;
+    EXPECT_EQ(BeInt64Id(0), ECJsonUtilities::JsonToId<BeInt64Id>(jsonDoc)) << jsonStr;
 
     ASSERT_EQ(SUCCESS, ParseJson(rapidJson, jsonStr));
-    EXPECT_EQ(BeInt64Id(uint64_t(-10)), ECJsonUtilities::JsonToId<BeInt64Id>(rapidJson)) << jsonStr;
+    EXPECT_EQ(BeInt64Id(0), ECJsonUtilities::JsonToId<BeInt64Id>(rapidJson)) << jsonStr;
 
     jsonStr = "3.14";  // Not a valid value for an Id, but failing the method is not worth the overhead. So the test documents the behavior of the API
     ASSERT_EQ(SUCCESS, ParseJson(jsonDoc, jsonStr));
-    EXPECT_EQ(BeInt64Id(3), ECJsonUtilities::JsonToId<BeInt64Id>(jsonDoc)) << jsonStr << " floating numbers are rounded";
+    EXPECT_EQ(BeInt64Id(0), ECJsonUtilities::JsonToId<BeInt64Id>(jsonDoc)) << jsonStr << " floating numbers are given defaultValue of 0.";

```

Comment:
I decided to make the JsonToId function more strict and return invalidIds for these types of numbers. It seemed more correct to me to turn a 3.14 into a 0 when trying to convert it to an ID, same with the string "3.14".

This all originated from me starting to hit RAPIDJSON_ASSERTS when calling m_val->GetUint64 on something like 3.14 when its a BeRapidJsonValue. This is because originally i was just checking that the value `isNumeric()` before calling `GetUint64()` on it. I temporarily switched over to `IsUint64` before calling `GetUint64` but I was worried that was too restrictive. which caused me to add some more test cases to this specific test where I casted to double,float,int64,int32. This also led to the addition of IsLosslessUint64 which takes double or float.